pennylane/qchem/convert_openfermion.py

```diff
@@ -45,96 +45,66 @@ def _import_of():
-    >>> pl_op = from_openfermion_qubit(of_op)
-    >>> print(pl_op)
-        0.5 * a⁺(0) a(2)
-        + 1.0 * a(0) a⁺(2)
-    """
-    coeffs, pl_ops = _openfermion_to_pennylane(of_op, tol=tol)
-    pl_term = qml.ops.LinearCombination(coeffs, pl_ops)
-
-    if "format" in kwargs:
-        if kwargs["format"] == "Sum":
-            return qml.dot(*pl_term.terms())
-        if kwargs["format"] != "LinearCombination":
-            f = kwargs["format"]
-            raise ValueError(f"format must be a Sum or LinearCombination, got: {f}.")
-
-    return pl_term
-
-
 def from_openfermion(openfermion_op, tol=1e-16):
     r"""Convert OpenFermion
     `FermionOperator <https://quantumai.google/reference/python/openfermion/ops/FermionOperator>`__
-    object to PennyLane :class:`~.fermi.FermiWord` or :class:`~.fermi.FermiSentence` objects.
+    and `QubitOperator <https://quantumai.google/reference/python/openfermion/ops/QubitOperator>`__
+    objects to PennyLane :class:`~.fermi.FermiWord` or :class:`~.fermi.FermiSentence` or
+    :class:`~.LinearCombination` objects.
 
     Args:
-        openfermion_op (FermionOperator): OpenFermion fermionic operator
+        openfermion_op (FermionOperator, QubitOperator): OpenFermion operator
+        wires (.Wires, list, tuple, dict): Custom wire mapping used to convert the external qubit
+            operator to a PennyLane operator.
+            For types ``Wires``/list/tuple, each item in the iterable represents a wire label
+            for the corresponding qubit index.
+            For type dict, only int-keyed dictionaries (for qubit-to-wire conversion) are accepted.
+            If ``None``, the identity map (e.g., ``0->0, 1->1, ...``) will be used.
         tol (float): tolerance for discarding negligible coefficients
 
     Returns:
-        Union[FermiWord, FermiSentence]: the fermionic operator object
+        Union[FermiWord, FermiSentence, LinearCombination]: PennyLane operator
 
     **Example**
 
-    >>> from openfermion import FermionOperator
-    >>> openfermion_op = 0.5 * FermionOperator('0^ 2') + FermionOperator('0 2^')
-    >>> pl_op = from_openfermion(openfermion_op)
+    >>> from openfermion import FermionOperator, QubitOperator
+    >>> of_op = 0.5 * FermionOperator('0^ 2') + FermionOperator('0 2^')
+    >>> pl_op = from_openfermion(of_op)
     >>> print(pl_op)
         0.5 * a⁺(0) a(2)
         + 1.0 * a(0) a⁺(2)
+
+    >>> of_op = QubitOperator('X0', 1.2) + QubitOperator('Z1', 2.4)
+    >>> pl_op = from_openfermion(of_op)
+    >>> print(pl_op)
+    1.2 * X(0) + 2.4 * Z(1)
     """
-    try:
-        import openfermion
-    except ImportError as Error:
-        raise ImportError(
-            "This feature requires openfermion. "
-            "It can be installed with: pip install openfermion"
-        ) from Error
+    openfermion = _import_of()
 
-    typemap = {0: "-", 1: "+"}
+    if isinstance(openfermion_op, openfermion.FermionOperator):
+        typemap = {0: "-", 1: "+"}
```

Comment:
We can put this conversion in a separate function and call that function here, as is being done for qubit operators.

Reply:
This version is simpler and better than having a function. Any reason a function is better?

Reply:
I suggested as that will make things consistent but we can leave it this way.